src/components/look-controls.js

```diff
@@ -242,5 +274,13 @@ module.exports.Component = registerComponent('look-controls', {
 
   onTouchEnd: function () {
     this.touchStarted = false;
+  },
+
+  onPointerLockChange: function () {
+    this.pointerLocked = !!(document.pointerLockElement || document.mozPointerLockElement);
+  },
+
+  onPointerLockError: function () {
+    this.pointerLocked = false;
```

Comment:
should probably also add

```js
  this.pointerLockedDenied = true;
```

and guard against that for the rest of the page session (until a browser reload).

I did [a bunch of tests](https://people.mozilla.org/~cwiemeersch/pointerlockerror-bug.html).

I discovered that Chrome actually fires `fullscreenerror` if you explicitly deny the Pointer Lock permission prompt.
Unfortunately, Firefox does not. (I filed a [platform bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1252749) with the test case.) A hacky workaround I can think of is to add some `setTimeout` to wait a few seconds and then assume the user rejected it.

I discovered that the `fullscreenerror` event _does_ get fired in Firefox - but only when trying to call `requestPointerLock` outside a user-initiated event.